### ISSUE TITLE
Reconfigure stale-bot GitHub action

### ISSUE THREAD

**Issue by waldyrious** [tldr-pages/tldr-lint#249]
I'm not sure if this is a new behavior introduced with the upgrade of the Stale action from v3 to to v8 (done by @kbdharun in #245), or if it had always been the case since its introduction (done by @owenvoke in #29), but we need to reconfigure the Stale action so that it doesn't close issues simply because they are stale.

The current behavior is especially bad because the stale-adding action is not accompanied by a comment, so even people watching this repo or the specific issues don't get a notification (I don't think I got one).

In fact, maybe it shouldn't even close PRs that are stale, but simply remind people of them.

Affected issues, which have recently been closed by stale bot (and need to be reopened): 

- #56
- #76
- #77
- #83
- #135
- #197
- #211
- #210 (already reopened)
- #93 (already reopened)

**Comment by waldyrious** [comment]
Note: #26 was marked as stale, but wasn't closed yet; I assume it's just a matter of time if we don't change the bot configuration.

**Comment by owenvoke** [comment]
@waldyrious, as far as I remember, the original way it was configured was to ignore issues. It would only close PRs. However, I can't find any point where this actually changed, so maybe it has always been this way (unintentionally). 🤔

I have opened https://github.com/tldr-pages/tldr-lint/pull/258 to work on these changes.

I agree that perhaps we should only remind people about PRs, and not close them. 🤔 Let me know if I should configure Stale Bot in this way.

**Comment by SethFalco** [comment]
Mostly regurgitating what others have said, but just want to push my thoughts forward.

I actually think we shouldn't have stale bot at all. It's just really disrespectful to contributors that invested their time in the project, just to have it thrown away because _maintainers_ didn't review it.

Even if we were to get out of our way to use it, we could add a label or comment once it's stale, but not automatically close issues _because_ it's stale. But I just don't see the value in it that, tbh.

The only time I'd appreciate automatic closing, is if the issue is both stale and had a label applied by a human that suggests we need further information from OP, but they aren't responsive. Like `Question` or `Cannot Reproduce`, where the OP is no longer responsive, and so we can't get the context to approve the change.

An issue/PR should be resolved somehow, even if the solution is `WONTFIX`. I do appreciate dropping a comment asking the OP if the issue/PR is still relevant, and to close is if it's been resolved incidentally. Sometimes things get resolved on their own. However, it should be a human action that results in closing it.

There is _no_ reason to close the issue automatically, unless a human has reviewed and determined that they specifically don't want it as part of the project or that the issue/PR is somehow flawed.

* WONTFIX: If the idea/PR is being rejected.
* Help Wanted: If it's a good idea, but as a maintainer, you would rather not invest your time in it.
* Cannot Reproduce: If you can't reproduce the error.
* etc

I think there was a conversation in Matrix about it as well, but I can't find it. :thinking:

---

Can someone explain what was our motivation for using this action in the first place? :thinking:
Before we consider reconfiguring it, I'd like to know what our motivation for using it is. What are we trying to achieve? Because the PR adding it, has no explanation and links to no external conversations.

* https://github.com/tldr-pages/tldr-lint/pull/29

**Comment by owenvoke** [comment]
Yeah, I too am not a huge fan of the stale approach. I think it was to improve visibility, but it tends to reduce it a lot of the time.

In terms of the addition to here, I think I originally added it to match the core repositorys stale bot (for consistency) (see https://github.com/tldr-pages/tldr/pull/4143, which was an update to the original Probot functionality introduced with https://github.com/tldr-pages/tldr/pull/1360 and https://github.com/tldr-pages/tldr/issues/1350).

**Comment by SethFalco** [comment]
Hmm, alright, based on the motivation given on tldr-pages/tldr, I can appreciate it more.
Everyone here more or less agrees that issues/PRs shouldn't be closed automatically, so I'll shelve that.

Imo we shouldn't even add the `stale` label either though. :thinking: 

Just a comment reminding maintainers to resolve the issue/PR one way or another. If the goal is to remind people, then all we need is the reminder, nothing else.

* If we asked OP a question, and they aren't responsive, we can close it.
* If we were waiting on OP to make fixes but aren't responsive, we can take over the PR.
* If we were waiting for more feedback from other maintainers but no one did, manually request reviews, share it in chat, or just merge it depending on the circumstances.

The label isn't required to adhere to the motivation, which is to ping individuals subscribed to the issue/PR.

I also don't think it's that practical, it just makes the issue/PR history a bit noisy. Even if someone wanted to get out of their way to tackle older issues, they'd probably go by date, not filter by the `stale` label. :thinking: 

Consider this PR, where more than half the activity is regarding stale bot:

![image](https://github.com/tldr-pages/tldr-lint/assets/22801583/88e38a0d-5144-49cb-91f2-b2ff080f9ccb)

**Comment by kbdharun** [comment]
Sorry for chiming in late, this ping got lost in my endless abyss of notifications 😅 .

> I'm not sure if this is a new behavior introduced with the upgrade of the Stale action from v3 to to v8 (done by @kbdharun in https://github.com/tldr-pages/tldr-lint/pull/245), or if it had always been the case since its introduction (done by @owenvoke in https://github.com/tldr-pages/tldr-lint/pull/29), but we need to reconfigure the Stale action so that it doesn't close issues simply because they are stale.

This behavior wasn't set/triggered by the changes in my PR as the version bump was a backend and Node version update (with additional improvements to the action). This closing behavior has always been the case since its introduction as you can see below:

```
   # Number of days of inactivity before an issue or PR is considered stale
        days-before-stale: 15
        # Number of days of inactivity before a stale issue or PR is closed
        days-before-close: 30
        # Issues or PRs with these labels will never be considered stale
        exempt-pr-labels: up for grabs
        # Label to use when marking an issue or PR as stale
        stale-pr-label: waiting
        # Comment to post when marking an issue or PR as stale
        stale-pr-message: |
          Hi all! This thread has not had any recent activity.
          Are there any updates? Thanks!
        # Comment to post when closing a stale issue or PR
        close-pr-message: |
          Hi everyone.
          This thread is being closed as there was no response to the previous prompt.
          However, please leave a comment whenever you're ready to resume,
          so the thread can be reopened. Thanks again!
```

> The current behavior is especially bad because the stale-adding action is not accompanied by a comment, so even people watching this repo or the specific issues don't get a notification (I don't think I got one).

Yeah

> Can someone explain what was our motivation for using this action in the first place? 🤔

I think it was so that inactive issues/PRs aren't open for a long time (see https://ossinsight.io/analyze/tldr-pages/tldr for instance). 

**I strongly agree that stale actions aren't necessary for both PRs and issues as many good PRs and issues are closed in most cases without us getting a notification. So it is better for us to manually intervene and close old issues rather than it being closed automatically both here and in the main repository.**

I agree with @SethFalco's comment too above, it kind of spams comments in major PRs waiting for a maintainer rather than the author in most cases.

I will open a PR to remove it in both repos.
